cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -33,7 +33,9 @@ import (
 )
 
 const (
-	debugFormat                    = "%s (min: %d, max: %d, replicas: %d)"
+	capiMachineDeleteAnnotationKey = "cluster.x-k8s.io/delete-machine"
```

Comment:
please don't instantiate this annotations here, just use the same code than upstream https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go#L56-L68

The only thing that needs to be instantiated statically is the delete annotation for mapi, see https://github.com/openshift/kubernetes-autoscaler/pull/230/files#diff-8266b9d377a9df07645393a09139352c01636b7c84b6678586245e72c69dfe45R38

Reply:
To clarify, I'm not proposing to switch over anything right now. All we need is now upstream which dynamically sets the group in the annotations via the get*annotationKey. https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go#L56-L68

So we need to drop everything else. The only skew we need is to have static `oldMachineDeleteAnnotationKey = "machine.openshift.io/cluster-api-delete-machine" because it's what mao uses and has a different suffix (so the dynamic get*annotation won't work for mapi in this case.).

Now, I'd suggest to eventually update mao to migrate the annotation to match upstream suffix (delete-machine) so eventually you don't need to carry this static annotation here.